File: Camera/Model/Photo.cs
using System;
using Newtonsoft.Json;

namespace Camera.Model
{
    public class Photo
    {
        [JsonIgnore]
        [AutoIncrement,PrimaryKey]
        public int Id { get; set; }

        [JsonProperty(PropertyName = "likes")]
        public int Likes { get; set; }


        [JsonProperty(PropertyName = "_id")]
        public string ServerId { get; set; }

        [JsonProperty(PropertyName = "thumbnail_url")]
        public string ThumbnailPath { get; set; }
        [JsonProperty(PropertyName = "full_url")]
        public string FullPath { get; set; }
        [JsonProperty(PropertyName = "root_url")]
        public string RootUrl { get; set; }

        [JsonProperty(PropertyName = "creation_time")]
        public DateTime CreationTime { get; set; }

        [JsonIgnore] 
        public string EventCode { get; set; }
    }
}

File: camera/Model/Photo.m
//
// Created by Owen Evans on 27/04/13.
// Copyright (c) 2013 Owen Evans. All rights reserved.
//
// To change the template use AppCode | Preferences | File Templates.
//


#import "Photo.h"


@implementation Photo {

}
-(void)setValue:(id)value forUndefinedKey:(NSString *)key {
    if(![key isEqualToString:@"photos"]){
        DDLogError(@"unable to set key %@ from value %@", key,value);
    }
}


- (NSURL *)fullURL {
    return [NSURL URLWithString:[NSString stringWithFormat:@"%@%@",[self rootUrl],[self fullPath]]];

}

+ (NSDictionary *)JSONKeyPathsByPropertyKey {
    DDLogInfo(@"getting key paths");
    return [[super JSONKeyPathsByPropertyKey] mtl_dictionaryByAddingEntriesFromDictionary: @{
            @"creationTime": @"creation_time",
            @"fullPath":@"full_url",
            @"thumbnailPath":@"thumbnail_url",
            @"rootUrl":@"root_url"
    }];
}

- (NSURL *)thumbnailURL {
    return [NSURL URLWithString:[NSString stringWithFormat:@"%@%@",[self rootUrl],[self thumbnailPath]]];
}

@end